src/lib/helpers.ts

```diff
@@ -344,3 +356,551 @@ export function parseResponseAPIVersion(response: Response) {
+      }
+    } catch (error) {
+      warnOnBrokenImplementation('getPublicKey()', error as Error)
+    }
+  }
+
+  // L3 says this is required, but browser and webview support are still not guaranteed.
+  let responseAuthenticatorData: string | undefined
+  if (typeof response.getAuthenticatorData === 'function') {
+    try {
+      responseAuthenticatorData = bufferToBase64URLString(response.getAuthenticatorData())
+    } catch (error) {
+      warnOnBrokenImplementation('getAuthenticatorData()', error as Error)
+    }
+  }
+
+  return {
+    id,
+    rawId: bufferToBase64URLString(rawId),
+    response: {
+      attestationObject: bufferToBase64URLString(response.attestationObject),
+      clientDataJSON: bufferToBase64URLString(response.clientDataJSON),
+      transports,
+      publicKeyAlgorithm: responsePublicKeyAlgorithm,
+      publicKey: responsePublicKey,
+      authenticatorData: responseAuthenticatorData,
+    },
+    type,
+    clientExtensionResults: credential.getClientExtensionResults(),
+    // authenticatorAttachment: toAuthenticatorAttachment(credential.authenticatorAttachment),
+  }
+}
+
+/**
+ * Attempt to intuit _why_ an error was raised after calling `navigator.credentials.create()`
+ */
+export function identifyRegistrationError({
+  error,
+  options,
+}: {
+  error: Error
+  options: CredentialCreationOptions
+}): WebAuthnError | Error {
+  const { publicKey } = options
+
+  if (!publicKey) {
+    throw Error('options was missing required publicKey property')
+  }
+
+  if (error.name === 'AbortError') {
+    if (options.signal instanceof AbortSignal) {
+      // https://www.w3.org/TR/webauthn-2/#sctn-createCredential (Step 16)
+      return new WebAuthnError({
+        message: 'Registration ceremony was sent an abort signal',
+        code: 'ERROR_CEREMONY_ABORTED',
+        cause: error,
+      })
+    }
+  } else if (error.name === 'ConstraintError') {
+    if (publicKey.authenticatorSelection?.requireResidentKey === true) {
+      // https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred (Step 4)
+      return new WebAuthnError({
+        message:
+          'Discoverable credentials were required but no available authenticator supported it',
+        code: 'ERROR_AUTHENTICATOR_MISSING_DISCOVERABLE_CREDENTIAL_SUPPORT',
+        cause: error,
+      })
+    } else if (publicKey.authenticatorSelection?.userVerification === 'required') {
+      // https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred (Step 5)
+      return new WebAuthnError({
+        message: 'User verification was required but no available authenticator supported it',
+        code: 'ERROR_AUTHENTICATOR_MISSING_USER_VERIFICATION_SUPPORT',
+        cause: error,
+      })
+    }
+  } else if (error.name === 'InvalidStateError') {
+    // https://www.w3.org/TR/webauthn-2/#sctn-createCredential (Step 20)
+    // https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred (Step 3)
+    return new WebAuthnError({
```

Comment:
I also don't think this applies for MFA.

Reply:
Yup it doesn't